deploy/template.yaml

```diff
@@ -1247,19 +1297,31 @@ Resources:
   JourneyEngineStepFunction:
     Type: AWS::Serverless::StateMachine
     Properties:
+      Name: !Sub JourneyEngineStepFunction-${AWS::StackName}
       DefinitionUri: journeyEngineStepFunction.asl.json
       DefinitionSubstitutions:
-        IPVProcessJourneyEventFunctionArn: !Ref IPVProcessJourneyEventFunction.Alias
-        CheckExistingIdentityFunctionArn: !Ref CheckExistingIdentityFunction.Alias
-        BuildCriOauthRequestFunctionArn: !Ref BuildCriOauthRequestFunction.Alias
-        BuildClientOauthResponseFunctionArn: !Ref BuildClientOauthResponseFunction.Alias
-        CheckGpg45ScoreFunctionArn: !Ref CheckGpg45ScoreFunction.Alias
-        EvaluateGpg45ScoresFunctionArn: !Ref EvaluateGpg45ScoresFunction.Alias
-        CallTicfCriLambdaArn: !Ref CallTicfCriFunction.Alias
-        CallDcmawAsyncCriLambdaArn: !Ref CallDcmawAsyncCriFunction.Alias
-        StoreIdentityLambdaArn: !Ref StoreIdentityFunction.Alias
-        ResetSessionIdentityFunctionArn: !Ref ResetSessionIdentityFunction.Alias
-        CheckCoiFunctionArn: !Ref CheckCoiFunction.Alias
+        IPVProcessJourneyEventFunctionArn: !Ref IPVProcessJourneyEventFunction.Version
+        CheckExistingIdentityFunctionArn: !Ref CheckExistingIdentityFunction.Version
+        BuildCriOauthRequestFunctionArn: !Ref BuildCriOauthRequestFunction.Version
+        BuildClientOauthResponseFunctionArn: !Ref BuildClientOauthResponseFunction.Version
+        CheckGpg45ScoreFunctionArn: !Ref CheckGpg45ScoreFunction.Version
+        EvaluateGpg45ScoresFunctionArn: !Ref EvaluateGpg45ScoresFunction.Version
+        CallTicfCriLambdaArn: !Ref CallTicfCriFunction.Version
+        CallDcmawAsyncCriLambdaArn: !Ref CallDcmawAsyncCriFunction.Version
+        StoreIdentityLambdaArn: !Ref StoreIdentityFunction.Version
+        ResetSessionIdentityFunctionArn: !Ref ResetSessionIdentityFunction.Version
+        CheckCoiFunctionArn: !Ref CheckCoiFunction.Version
```

Comment:
alarms for all the above functions should be configured in the state machine as they are associated with a state machine e.g. in bravo - https://github.com/govuk-one-login/di-id-reuse-storage/blob/3d040e9f94162790a1ad57d0ee862e2a44207901/src/infra/main/template.yaml#L587
although may need double checking as they seem to also be called via api not just via state machine